docs/building-apps/building-static-apps.rst

```diff
@@ -0,0 +1,73 @@
+Building 100% static MPI applications
+=====================================
+
+.. caution:: Fully static linking is not for the meek, and it is not
+   recommended.  But it is possible, with some caveats.
+
+#. You must have static libraries available for *everything* that your
+   program links to.  This includes Open MPI; you must have used the
+   ``--enable-static`` option to Open MPI's ``configure`` or otherwise
+   have available the static versions of the Open MPI libraries (note
+   that Open MPI static builds default to including all of its plugins
+   *in* its libraries |mdash| as opposed to having each plugin in its
+   own dynamic shared object file.  So *all* of Open MPI's code will
+   be contained in the static libraries |mdash| even what are normally
+   contained in Open MPI's plugins).  Note that some popular Linux
+   libraries do not have static versions by default (e.g., libnuma),
+   or require additional RPMs to be installed to get the equivalent
+   libraries.
+
+#. Open MPI must have been built without a memory manager.  This means
+   that Open MPI must have been configured with the
+   ``--without-memory-manager`` flag.  This is irrelevant on some
+   platforms for which Open MPI does not have a memory manager, but on
+   some platforms it is necessary (Linux).  It is harmless to use this
+   flag on platforms where Open MPI does not have a memory manager.
+   Not having a memory manager means that Open MPI's
+   ``mpi_leave_pinned`` behavior for OS-bypass networks such as
+   InfiniBand will not work.
+
+#. On some systems (Linux), you may see linker warnings about some
+   files requiring dynamic libraries for functions such as
+   ``gethostname`` and ``dlopen``.  These are ok, but do mean that you
+   need to have the shared libraries installed.  You can disable all
+   of Open MPI's ``dlopen`` behavior (i.e., prevent it from trying to
```

Comment:
I think we should move this earlier, with some more context.  Almost a "what are you trying to do?" intro.  If they just want us to not build components, --disable-mca-dso is really the best option.  There are a couple of places that we use dlopen() and dlsym() outside of MCA (for leave pinned, for example), although I don't know that we're as good as we should be about following --disable-dlopen there.

Reply:
I think I want to talk to you verbally about your two notes about content; it would probably be easier / quicker to come to consensus on the phone.